docs/guides/advanced-experimentation/anonymous-explainer.md

```diff
@@ -4,110 +4,90 @@ sidebar_position: 5
+On the assignment page ... (finish this once we have a screenshot in QA)
 
-```sql
-SELECT
-    t1.*
-FROM anon_visitor_to_user_mapping t1
-     LEFT JOIN anon_visitor_to_user_mapping t2
-           ON t1.user_id = t2.user_id
-               -- Checking to see if there's an overlapping attribution window for the SAME user_id with a different anonymous id
-               -- If an overlap exists, there is the possibility of double counting fact events
-               AND t1.anonymous_id <> t2.anonymous_id
-               AND t1.ts_end_window > t2.ts_start_window
-               AND t2.ts_end_window > t1.ts_start_window
-WHERE t2.user_id IS NOT NULL
-```
+## Data Modeling Considerations
 
-Once this model is built, it can be joined to any fact table within the data warehouse. It should be joined onto these fact tables by User ID wherever a fact event’s timestamp is between a given user’s `ts_start_window` and `ts_end_window`. By doing this, all fact tables at the user level can now have an inferred Anonymous ID. This inferred Anonymous ID can then be used by Eppo to link Assignment SQL definitions at the Anonymous ID level to these fact tables.
+The approach above assumes you have relatively dense assignment data. That is, if an anonymous ID is later associated with a user, there will be a corresponding event in the assignment logs. This can be error prone. If a user is assigned on the initial landing page and then converts on a different page, it's easy to imagine the assignment log not having a record with that association. To solve this, we can simply do a one time join between the raw assignment table and another table with the full set of timestamped associations:
 
 ```sql
-SELECT
-   facts.ts
-   , mapping.anonymous_id
-   , facts.user_id
-FROM fact_table as facts
-LEFT JOIN anon_visitor_to_user_mapping AS mapping
-  ON facts.user_id = mapping.user_id
-    AND facts.ts BETWEEN
-      mapping.ts_start_window AND
-      mapping.ts_end_window
+WITH timeboxed_associations AS (
+  SELECT
+        user_id
+      , anonymous_id
+      , lag(ts) OVER (PARTITION BY user_id ORDER BY ts) as valid_from
+      , lead(ts) OVER (PARTITION BY user_id ORDER BY ts) as valid_to
+  FROM user_anonymous_associations
+)
+
+SELECT 
+      a.anonymous_id
+    , ta.user_id
+    , a.experiment
+    , a.variant
+    , a.timestamp
+   FROM assignments a 
+   LEFT JOIN timeboxed_associations ta
+     ON a.anonymous_id = ta.anonymous_id
+    AND ta.valid_from <= a.timestamp
```

Comment:
Should this have an `OR ta.valid_from IS NULL`? Thinking about the case where an anonymous ID is the first one for a given `user_id`, so the lag function would return `NULL`. Ditto for `valid_to`

Reply:
yep definitely